### ISSUE TITLE
Anyway to customize the Wave interface?

### ISSUE THREAD

**Issue by aeluviternis** [wavetermdev/waveterm#636]
As the title suggests. 

I've recently been introduced to Wave and have quickly fallen in love with it. It truly is a wonderful piece of work, and please allow me to express my sincere gratitude for the efforts put into developing and open sourcing it.

I'm curious if there's a possibility to customize the Wave interface, such as implementing command code highlighting and custom themes? I've come across some discussions regarding this, but I'm not sure if anyone has actually implemented these features yet.

Best regards!

**Comment by sawka** [comment]
Yes, we're adding custom themes -- specifically for the terminal interfaces, you'll be able to set custom colors, fonts, and backgrounds.

We don't have a way to customize command highlighting, but we're open to suggestions.  One thought is to replace our cmdinput textarea with a small version of the Monaco editor (the VSCode editor) which will allow us more flexibility for auto-complete and code highlighting.

**Comment by esimkowitz** [comment]
Yeah syntax highlighting would be super cool! Check out this issue: #585

**Comment by yoghurtlee-thu** [comment]
Sounds great. Another personal suggestion: Zsh (Oh-my-Zsh, actually) could be integrated directly into the interface, which would allow the use of powerful Zsh plugins such as syntax highlighting and autosuggestion. (I'm not an expert in this area, so please forgive any foolish mistakes in my expression and suggestions).

**Comment by djdanielsson** [comment]
Oh-my-Zsh is great and there is also https://github.com/JanDeDobbeleer/oh-my-posh that might be worth looking at supporting.

**Comment by DannyBen** [comment]
I am just starting with Wave, and found two places for colors:

1. Right clicking the tab and choosing background
2. Clicking the gear of a widget and choosing theme

None of these contain any option for a light theme. Is this indeed the case?

**Comment by esimkowitz** [comment]
Correct, we don't yet support a proper light theme

**Comment by ed-code-dev** [comment]
well, then back to warp...

**Comment by esimkowitz** [comment]
haha stay tuned, it's coming, we just have been working on stability recently so haven't gotten around to it

**Comment by amarBitMan** [comment]
@esimkowitz  
I don't find the light theme in current list, how can I add it or is it not possible yet ??

**Comment by esimkowitz** [comment]
@amarBitMan there isn't currently a light theme, however it should be somewhat straightforward to add, you can make another version of frontend/app/theme.scss with light mode values. That should get you a good part of the way there. Probably some hardcoded values though that will need to be refactored

**Comment by amarBitMan** [comment]
> [@amarBitMan](https://github.com/amarBitMan) there isn't currently a light theme, however it should be somewhat straightforward to add, you can make another version of frontend/app/theme.scss with light mode values. That should get you a good part of the way there. Probably some hardcoded values though that will need to be refactored

Could you please share any steps or examples on how to do this, if possible? I’m fairly new to Wave and still getting familiar with the setup process.

**Comment by esimkowitz** [comment]
There are some basic background theme overrides specified in https://github.com/wavetermdev/waveterm/blob/main/pkg/wconfig/defaultconfig/presets.json but they're not a true light mode, i.e. text colors won't change. We don't have any steps or examples defined, I'm not super involved in this project anymore but if you put something together I'd be happy to review it

**Comment by BasicOneCruisider** [comment]
hi, here is my github, with full of theme , and some widget, Enjoy 
https://github.com/BasicOneCruisider/wavetermwidgets

**Comment by Gabriel-p** [comment]
Where should the `termthemes.json` file be stored? The [Customizatio](https://docs.waveterm.dev/customization) page only says:

> If you add your own termthemes.json file in the config directory, you can also add your own custom terminal themes (just follow the same format).

**Comment by esimkowitz** [comment]
The configuration directory is documented here: https://docs.waveterm.dev/config

**Comment by Gabriel-p** [comment]
Thanks @esimkowitz but the docs are either old or they don't apply to Snap installations. For starters I have no `~/.config/waveterm/` folder, everything is located inside of `~/snap/waveterm/`. There are several folders within that one named `themes` but no `termthemes.json` file anywhere

**Comment by esimkowitz** [comment]
Ah, try this command then: https://docs.waveterm.dev/wsh-reference#wavepath

That should show you where it is, it's sandboxed for Snap. The documentation used to point this out, not sure when that changed...

**Comment by Gabriel-p** [comment]
I don't understand how the keywords in the `termthemes.json` file work. I'm trying to set a background similar to Solarized Light (#fdf6e3) and this is what I get:

<img width="1512" height="799" alt="Image" src="https://github.com/user-attachments/assets/fb4737fd-4f6b-401a-88d5-2f2889098a5e" />

**Comment by mcfearsome** [comment]
Another request for status on light themes? Attempting to create my own I get the same problem as @Gabriel-p, this is going to keep me from switching and I'd like to give it a fair try
